sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuTaskMetrics.scala

```diff
@@ -289,11 +290,26 @@ class GpuTaskMetrics extends Serializable {
     }
   }
 
+  private def timeIt[A](timer: NanoSecondAccumulator,
+                        range: NvtxId,
+                        color: NvtxColor,
+                        f: => A): A = {
+    val start = System.nanoTime()
+    withResource(new NvtxRangeWithDoc(range, color)) { _ =>
```

Comment:
nit: if you make it a case class or add a companion with factory apply manually 
```suggestion
    withResource(NvtxRangeWithDoc(range, color)) { _ =>
```